development-kubeconfig.yaml

```diff
@@ -0,0 +1,24 @@
+apiVersion: v1
+clusters:
+- cluster:
+    # Since minikube generates self-signed certificate that doesn't include
+    # hostname pccda003.cern.ch tls hostname verify has to be skipped
+    insecure-skip-tls-verify: true
+    #certificate-authority: /home/hahirvon/.minikube/ca.crt
+    server: https://pccda003.cern.ch:8443
```

Comment:
This is personal configuration that people would have to touch... Can you make it work without `development-kubeconfig.yaml`?  Currently, without this file, I get:

```console
$ rm development-kubeconfig.yaml
$ reana-cluster -f reana-cluster-dev.yaml init
[INFO] Validating REANA cluster specification file: /home/simko/private/src/reana-cluster/reana-cluster-dev.yaml
[INFO] /home/simko/private/src/reana-cluster/reana-cluster-dev.yaml is a valid REANA cluster specification.
[INFO] Cluster type specified in cluster specifications file is 'kubernetes'
[ERROR] 'NoneType' object has no attribute 'cluster_type'
Traceback (most recent call last):
  File "/home/simko/private/src/reana-cluster/reana_cluster/cli/cluster.py", line 108, in init
    .format(cluster=backend.cluster_type,
AttributeError: 'NoneType' object has no attribute 'cluster_type'
```

Reply:
... actually I'm trying now with this change as well:

```console
$ git diff reana-cluster-dev.yaml 
diff --git a/reana-cluster-dev.yaml b/reana-cluster-dev.yaml
index fd8060a..6c1577d 100644
--- a/reana-cluster-dev.yaml
+++ b/reana-cluster-dev.yaml
@@ -1,6 +1,5 @@
 cluster:
   type: "kubernetes"
-  config: "./development-kubeconfig.yaml"
   config_context: "minikube"
   version: "v1.6.4"
   url: "http://localhost"
```

Let's see...

Reply:
Yes it could be removed or at least cleaned up from personal configuration and commented out in `reana-cluster-dev`. If it will be commented out, I would also write a short description about key and it's usage.

Reply:
... so the local change propagation to containers is working fine 😄 -- see the changed `README.rst` file below that I touched on host OS -- but I see a mixup with mount directories:

```
reanauser@job-controller-2461563162-hwnjz:/code$ ls -l
total 80
-rw-r--r-- 1 reanauser reanauser   177 Apr  5  2017 AUTHORS.rst
-rw-r--r-- 1 reanauser reanauser   245 Apr  5  2017 CHANGES.rst
-rw-r--r-- 1 reanauser reanauser   889 Apr  5  2017 CONTRIBUTING.rst
-rw-r--r-- 1 reanauser reanauser 18009 Apr  5  2017 COPYING
-rw-r--r-- 1 reanauser reanauser  1182 Apr  5  2017 Dockerfile
-rw-r--r-- 1 reanauser reanauser  1230 May 15  2017 MANIFEST.in
-rw-r--r-- 1 reanauser reanauser  1045 Apr  5  2017 README.rst
drwxr-xr-x 1 reanauser reanauser  4096 Jul  6  2017 docs
-rw-r--r-- 1 reanauser reanauser  1025 Apr  5  2017 pytest.ini
drwxr-xr-x 1 reanauser reanauser  4096 Jan 16 18:15 reana-job-controller
drwxr-xr-x 1 reanauser reanauser  4096 Jul  6  2017 reana_job_controller
drwxr-xr-x 1 reanauser reanauser  4096 May 15  2017 reana_job_controller.egg-info
-rwxr-xr-x 1 reanauser reanauser  1370 May 25  2017 run-tests.sh
-rw-r--r-- 1 reanauser reanauser  1059 May  5  2017 setup.cfg
-rw-r--r-- 1 reanauser reanauser  3256 Jul  6  2017 setup.py
drwxr-xr-x 1 reanauser reanauser  4096 Jul  6  2017 tests
reanauser@job-controller-2461563162-hwnjz:/code$ ls -l reana-job-controller/
total 49
-rw-r--r-- 1 reanauser reanauser   177 Jan 25  2017 AUTHORS.rst
-rw-r--r-- 1 reanauser reanauser   245 Jan 25  2017 CHANGES.rst
-rw-r--r-- 1 reanauser reanauser   889 Feb  1  2017 CONTRIBUTING.rst
-rw-r--r-- 1 reanauser reanauser 18009 Jan 25  2017 COPYING
-rw-r--r-- 1 reanauser reanauser  1188 Jan 16 18:15 Dockerfile
-rw-r--r-- 1 reanauser reanauser  1230 May 15  2017 MANIFEST.in
-rw-r--r-- 1 reanauser reanauser  1051 Jan 24 10:29 README.rst
drwxr-xr-x 1 reanauser reanauser  4096 May 18  2017 docs
-rw-r--r-- 1 reanauser reanauser  1025 Jan 25  2017 pytest.ini
drwxr-xr-x 1 reanauser reanauser  4096 Jan 11 14:41 reana_job_controller
drwxr-xr-x 1 reanauser reanauser  4096 May 15  2017 reana_job_controller.egg-info
-rwxr-xr-x 1 reanauser reanauser  1370 May 15  2017 run-tests.sh
-rw-r--r-- 1 reanauser reanauser  1059 Jan 25  2017 setup.cfg
-rw-r--r-- 1 reanauser reanauser  3429 Jan 16 18:15 setup.py
drwxr-xr-x 1 reanauser reanauser  4096 May 15  2017 tests
```

I guess the following in `reana-cluster-dev.yaml`:

```
path: "/devcomponents/reana-job-controller:/code/reana-job-controller"
```

should rather be:

```
path: "/devcomponents/reana-job-controller:/code/"
```

Reply:
Correct :man_facepalming: